components/post_priority/post_priority_picker_overlay.tsx

```diff
@@ -1,56 +1,146 @@
+            <FormattedMessage
+                id='shortcuts.msgs.formatting_bar.post_priority'
+                defaultMessage={'Message priority'}
+            />
+        ),
     });
-    const offset = pickerPosition(target(), show);
+
+    const handleClose = useCallback(() => {
+        setPickerOpen(false);
+        onClose();
+    }, [onClose]);
+
+    const {
+        x: pickerX,
+        y: pickerY,
+        reference: pickerRef,
+        floating: pickerFloating,
+        strategy: pickerStrategy,
+        context: pickerContext,
+    } = useFloating({
+        open: pickerOpen,
+        onOpenChange: setPickerOpen,
+        placement: 'top-start',
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offset({mainAxis: 4}),
+            flip({
+                fallbackPlacements: ['top'],
+            }),
+            shift({
+                padding: 16,
+            }),
+        ],
+    });
+
+    const {
+        getFloatingProps: getPickerFloatingProps,
+        getReferenceProps: getPickerReferenceProps,
+    } = useInteractions([
+        useClick(pickerContext),
+        useDismiss(pickerContext),
+        useRole(pickerContext),
+    ]);
 
     return (
-        <Overlay
-            show={show}
-            placement={'top'}
-            rootClose={true}
-            onHide={onHide}
-            target={target}
-            animation={false}
-        >
-            <PostPriorityPicker
-                settings={settings}
-                leftOffset={offset}
-                onApply={onApply}
-                topOffset={-7}
-                placement={'top'}
-                onClose={onHide}
-            />
-        </Overlay>
+        <>
+            <div
+                ref={tooltipRef}
+                {...getTooltipReferenceProps()}
+            >
+                <IconContainer
+                    ref={pickerRef}
+                    className={classNames({control: true, active: pickerOpen})}
+                    disabled={disabled}
+                    type='button'
+                    {...getPickerReferenceProps()}
+                >
+                    <AlertCircleOutlineIcon
+                        size={18}
+                        color='currentColor'
+                    />
+                </IconContainer>
+            </div>
+            <FloatingPortal id='root-portal'>
+                {pickerOpen && (
+                    <FloatingFocusManager
+                        context={pickerContext}
+                        modal={true}
+                        returnFocus={false}
+                    >
+                        <div
+                            ref={pickerFloating}
+                            style={{
+                                width: 'max-content',
+                                position: pickerStrategy,
+                                top: pickerY ?? 0,
+                                left: pickerX ?? 0,
+                                zIndex: 999,
```

Comment:
0/5 - Needing to set a z-index here seems weird since this is being portalled to the root level already. I would think that it's above the rest of the DOM simply because of its position in it